hls4ml/converters/pytorch/core.py

```diff
@@ -62,9 +62,13 @@ def parse_activation_layer(operation, layer_name, input_names, input_shapes, nod
             layer['activation'] = 'ThresholdedReLU'
             if layer['activ_param'] < 0:
                 raise Exception('negative threshold values not supported')
-
-        if hasattr(node, 'dim'):
+        if hasattr(class_object, 'dim'):
             layer['axis'] = class_object.dim
+            if layer['class_name'] == 'Softmax' and layer['axis'] is None:
+                layer['axis'] = -1
+            if 'IOType' in config:
```

Comment:
We have done these types of checks before, but now I think we should parse everything into the IR, and then backbends should check if the feature is supported. We can have optimizers for that. Check out feature_check.py in Vitis backend. We could add similar checks for all backends and move away from the parser (in a future PR)